doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -149,74 +148,71 @@ always the intended equality for a given structure.
+For instance, the tactic reflexivity can be used to solve a goal ``R n n`` whenever ``R``
 is an instance of a registered reflexive relation. However, the
 tactics that replace in a context ``C[]`` one term with another one
 related by ``R`` must verify that ``C[]`` is a morphism that respects the
-intended relation. Currently the verification consists in checking
+intended relation. Currently the verification consists of checking
 whether ``C[]`` is a syntactic composition of morphism instances that respects some obvious
 compatibility constraints.
 
+.. example:: Rewriting
 
-**Example 5 (Rewriting):**
-
-Continuing the previous examples, suppose that the user must prove
-``set_eq int (union int (union int S1 S2) S2) (f S1 S2)`` under the
-hypothesis ``H: set_eq int S2 (@nil int)``. It
-is possible to use the ``rewrite`` tactic to replace the first two
-occurrences of ``S2`` with ``@nil int`` in the goal since the
-context ``set_eq int (union int (union int S1 nil) nil) (f S1 S2)``,
-being a composition of morphisms instances, is a morphism. However the
-tactic will fail replacing the third occurrence of ``S2``  unless ``f``
-has also been declared as a morphism.
+   Continuing the previous examples, suppose that the user must prove
+   ``set_eq int (union int (union int S1 S2) S2) (f S1 S2)`` under the
+   hypothesis ``H : set_eq int S2 (@nil int)``. It
+   is possible to use the ``rewrite`` tactic to replace the first two
+   occurrences of ``S2`` with ``@nil int`` in the goal since the
+   context ``set_eq int (union int (union int S1 nil) nil) (f S1 S2)``,
+   being a composition of morphisms instances, is a morphism. However the
+   tactic will fail replacing the third occurrence of ``S2``  unless ``f``
+   has also been declared as a morphism.
 
 
 Adding new relations and morphisms
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-A parametric relation :g:`Aeq: forall (y1 : β1 ... ym : βm )`,
-:g:`relation (A t1 ... tn)` over :g:`(A : αi -> ... αn -> Type)` can be
-declared with the following command:
+.. cmd::  Add Parametric Relation (x1 : T1) ... (xn : Tk) : (A t1 ... tn) (Aeq t′1 ... t′m) {? reflexivity proved by refl} {? symmetry proved by sym} {? transitivity proved by trans} as @ident
 
-.. cmd::  Add Parametric Relation (x1 : T1) ... (xn : Tk) : (A t1 ... tn) (Aeq t′1 ... t′m ) {? reflexivity proved by refl} {? symmetry proved by sym} {? transitivity proved by trans} as @ident
-
-after having required the ``Setoid`` module with the ``Require Setoid``
-command.
+   This command declares a parametric relation :g:`Aeq: forall (y1 : β1 ... ym : βm)`,
+   :g:`relation (A t1 ... tn)` over :g:`(A : αi -> ... αn -> Type)`.
 
-The :g:`@ident` gives a unique name to the morphism and it is used
-by the command to generate fresh names for automatically provided
-lemmas used internally.
+   The :token:`@ident` gives a unique name to the morphism and it is used
```

Comment:
When using `:token:` you shouldn't put the `@`. The `@` is used inside `:n:` blocks to denote grammar tokens.